front/app/modules/commercial/customizable_homepage_banner/admin/CTASettings/index.tsx

```diff
@@ -2,59 +2,49 @@ import { Section, SubSectionTitle } from 'components/admin/Section';
 import { FormattedMessage } from 'utils/cl-intl';
 import React from 'react';
 import messages from '../messages';
-import {
-  TAppConfigurationSetting,
-  IAppConfigurationSettings,
-} from 'services/appConfiguration';
+import { IHomepageSettingsAttributes } from 'services/homepageSettings';
 import CTASignedOutSettings from './CTASignedOutSettings';
 import CTASignedInSettings from './CTASignedInSettings';
 import { CLErrors } from 'typings';
 
+// only these keys will be used in CTA settings
+export type BannerSettingKeyTypes = Extract<
+  IHomepageSettingsAttributes,
+  | 'banner_cta_signed_in_text_multiloc'
+  | 'banner_cta_signed_out_text_multiloc'
+  | 'banner_cta_signed_in_url'
+  | 'banner_cta_signed_out_url'
+  | 'banner_cta_signed_out_type'
+  | 'banner_cta_signed_in_type'
+>;
```

Comment:
Did you check? This type is `never`. What I think we want is 

```
export type BannerSettingKeyType = Extract<
  keyof IHomepageSettingsAttributes,
  | 'banner_cta_signed_in_text_multiloc'
  | 'banner_cta_signed_out_text_multiloc'
  | 'banner_cta_signed_in_url'
  | 'banner_cta_signed_out_url'
  | 'banner_cta_signed_out_type'
  | 'banner_cta_signed_in_type'
>;
```

Then in the props

`    settingKey: BannerSettingKeyType,`

Also note, it should be singular, it's only one type, not multiple types, so I changed it to BannerSettingKeyType (but details). :-)

Reply:
(already pushed because I made the changes while trying)